library/std/src/sys/hermit/net.rs

```diff
@@ -56,6 +56,11 @@ impl Socket {
         unimplemented!()
     }
 
+    pub fn connect(&self, addr: &SocketAddr) -> io::Result<()> {
+        let (addr, len) = addr.into_inner();
+        cvt_r(unsafe { netc::connect(self.as_raw_fd(), addr.as_ptr(), len) })
```

Comment:
This does not compile.

```suggestion
        cvt_r(|| unsafe { netc::connect(self.as_raw_fd(), addr.as_ptr(), len) })?;
        Ok(())
```

Reply:
Thanks for this catch. I have originally used `cvt` instead of the `cvt_r` here, but then realized that I don't know anything about hermit os and whether it can interrupt connect by a signal or not. So, finally I decided to leave everything as it was before. BTW, is there any documentation to read about connect on hermit?

Reply:
> is there any documentation to read about connect on hermit?

I am afraid not. There's the function definition in [`hermit_abi::connect`](https://docs.rs/hermit-abi/latest/hermit_abi/fn.connect.html) and there is [the actual implementation](https://github.com/search?q=repo%3Ahermit-os%2Fkernel+connect&type=code). It's not documented, though, and may change in the future. Currently, we mimic BSD sockets.